backend/src/services/webSocketService.ts

```
@@ -0,0 +1,234 @@
+export interface WebSocketUser {
+  userId: string;
+  socketId: string;
+  connectedAt: Date;
+  subscriptions: Set<string>;
+}
+
+export class WebSocketService {
+  private io: SocketIOServer;
+  private users: Map<string, WebSocketUser> = new Map();
+  private monitor: TransactionMonitor | null = null;
+  private healthCheckInterval: NodeJS.Timeout | null = null;
```

ESLint

[error] 'NodeJS' is not defined.

(no-undef)